[Split Build] Add option to create libtorch wheel and use it to build pytorch as a separate wheel

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126930
* #126590
* #126929
* #126813
* #126699
* #126826
* __->__ #126328

Creates an option to just build the libtorch portion of pytorch such that we have the necessary .so files.  Then it builds a torch package using the libtorch wheel. These options are enabled using ` BUILD_LIBTORCH_WHL` and `BUILD_PYTHON_ONLY`.

We run 

```
 BUILD_LIBTORCH_WHL=1 python setup.py install
python setup.py clean
BUILD_PYTHON_ONLY=1 python setup.py install
```

to produce

```
sahanp@devgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ls /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/torch/lib/                                                                                                                (pytorch-3.10) 
libshm.so*  libtorch_global_deps.so*  libtorch_python.so*
sahanp@devgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ldd build/lib/libtorch_python.so                                                                                                                                                                (pytorch-3.10) 
        linux-vdso.so.1 (0x00007ffdc2d37000)
        libtorch.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch.so (0x00007f539fe99000)
        libshm.so => /home/sahanp/pytorch/build/lib/libshm.so (0x00007f539fe90000)
        libcudnn.so.8 => /usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn.so.8 (0x00007f539e800000)
        libnvToolsExt.so.1 => /usr/local/cuda/lib64/libnvToolsExt.so.1 (0x00007f539e400000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f539e000000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f539fda5000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f539ebe5000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f539dc00000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f539fea0000)
        libtorch_cpu.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch_cpu.so (0x00007f5392400000)
        libtorch_cuda.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch_cuda.so (0x00007f5380000000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f539fd9e000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f539fd99000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f539fd94000)
        libc10.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libc10.so (0x00007f539eb07000)
        libmkl_intel_lp64.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_intel_lp64.so.2 (0x00007f537ec00000)
        libmkl_gnu_thread.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_gnu_thread.so.2 (0x00007f537ce00000)
        libmkl_core.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_core.so.2 (0x00007f5378800000)
        libomp.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/libomp.so (0x00007f539e707000)
        libcupti.so.12 => /usr/local/cuda/lib64/libcupti.so.12 (0x00007f5377e00000)
        libcudart.so.12 => /usr/local/cuda/lib64/libcudart.so.12 (0x00007f5377a00000)
        libc10_cuda.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libc10_cuda.so (0x00007f539ea6a000)
        libcusparse.so.12 => /usr/local/cuda/lib64/libcusparse.so.12 (0x00007f5368400000)
        libcufft.so.11 => /usr/local/cuda/lib64/libcufft.so.11 (0x00007f535ee00000)
        libcusolver.so.11 => /usr/local/cuda/lib64/libcusolver.so.11 (0x00007f534c800000)
        libcurand.so.10 => /usr/local/cuda/lib64/libcurand.so.10 (0x00007f5346200000)
        libcublas.so.12 => /usr/local/cuda/lib64/libcublas.so.12 (0x00007f533f800000)
        libcublasLt.so.12 => /usr/local/cuda/lib64/libcublasLt.so.12 (0x00007f531e800000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007f539ea63000)
        libnvJitLink.so.12 => /usr/local/cuda/lib64/libnvJitLink.so.12 (0x00007f531b800000)
sahanp@devgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ldd build/lib/libtorch_global_deps.so                                                                                                                                                           (pytorch-3.10) 
        linux-vdso.so.1 (0x00007ffc265df000)
        libmkl_intel_lp64.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_intel_lp64.so.2 (0x00007fa93fc00000)
        libmkl_gnu_thread.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_gnu_thread.so.2 (0x00007fa93de00000)
        libmkl_core.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_core.so.2 (0x00007fa939800000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fa940f05000)
        libcudart.so.12 => /usr/local/cuda/lib64/libcudart.so.12 (0x00007fa939400000)
        libnvToolsExt.so.1 => /usr/local/cuda/lib64/libnvToolsExt.so.1 (0x00007fa939000000)
        libgomp.so.1 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libgomp.so.1 (0x00007fa93fb07000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fa938c00000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fa940efe000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fa940ef9000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa940ff5000)
        librt.so.1 => /lib64/librt.so.1 (0x00007fa940ef2000)
        libstdc++.so.6 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libstdc++.so.6 (0x00007fa93921d000)
        libgcc_s.so.1 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libgcc_s.so.1 (0x00007fa93faec000)
        ```

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

atalman commented on setup.py:
```diff
@@ -220,7 +219,14 @@
+#   BUILD_LIBTORCH_WHL
+#      Builds libtorch.so and its dependencies as a wheel
+#
+#   BUILD_PYTORCH_USING_LIBTORCH_WHL
```
If we are not supporting this option ``BUILD_PYTORCH_USING_LIBTORCH_WHL`` yet should we be adding this code now ?

atalman commented on setup.py:
```diff
@@ -1356,6 +1403,22 @@ def main():
         "utils/model_dump/*.mjs",
     ]
 
+    if BUILD_PYTORCH_USING_LIBTORCH_WHL:
```
If this option ``BUILD_PYTORCH_USING_LIBTORCH_WHL`` is not supported yet, I think its better to add it when its ready.

PaliC replied:
For this PR I think it's reasonable given I'm not planning on merging this one without the next one on the stack. Also I'd rather not rewrite the dependency logic.

PaliC replied:
Though for next time you're right

atalman commented on setup.py:
```diff
@@ -1398,6 +1461,29 @@ def main():
+        for package in packages:
+            parts = package.split(".")
+            if parts[0] == "torch":
+                modified_packages.append("libtorch" + package[len("torch") :])
```
Here in place of "libtorch" perhaps would be better to use ``DEFAULT_PACKAGE_NAME``

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

kit1980 commented on setup.py:
```diff
@@ -1079,6 +1104,32 @@ def print_box(msg):
+    """
+    result = {}
+    for package in package_list:
+        # Split the package name by dots to handle subpackages or modules
+        parts = package.split(".")
+        # Check if the top-level package is 'torch'
+        if parts[0] == "torch":
+            # Replace 'torch' with 'libtorch' in the top-level package name
+            new_key = "libtorch" + package[len("torch") :]
+            result[new_key] = package
```
I think this whole thing is equivalent to 
```
if package.startswith("torch"):
    result["lib" + package] = package
```

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

ZainRizvi commented on setup.py:
```diff
@@ -1079,6 +1103,32 @@ def print_box(msg):
     print("-" * (size + 2))
 
 
+def rename_torch_packages(package_list):
```
does this get used in a future PR?

atalman commented on setup.py:
```diff
@@ -1079,6 +1103,32 @@ def print_box(msg):
 
+def rename_torch_packages(package_list):
+    """
+    Create a dictionary from a list of package names, renaming packages where
```
Why rename is necessary ? Maybe you can describe hi-level workflow of this build

PaliC replied:
It's no longer needed. We can just delete this.

PaliC committed:
Update on "[Split Build] Add option to create libtorch wheel"


Creates an option to just build the libtorch portion of pytorch such that we have the necessary .so files. 

Testing:
Running `BUILD_LIBTORCH_WHL=1 with-proxy python setup.py install` gets you  a library with the following files

```
sahanpdevgpu086 ~/pytorch (main)> ls /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/                                  (pytorch-3.10) 
libbackend_with_compiler.so*  libc10d_cuda_test.so*  libcaffe2_nvrtc.so*     libshm/          libtorchbind_test.so*  libtorch_cuda.so*         libtorch.so*
libc10_cuda.so*               libc10.so*             libjitbackend_test.so*  libshm_windows/  libtorch_cpu.so*       libtorch_global_deps.so*
```

Future Work:
We can trim this down a bit more, but for now this is enough to start building using the libtorch build.

[ghstack-poisoned]

kit1980 reviewed (changes requested):
Please have CMakeLists.txt formatting changes in a separate PR.

kit1980 reviewed (commented):
Why is third_party/cpp-httplib submodule updated?

atalman commented on setup.py:
```diff
@@ -210,6 +217,34 @@
+
+
+if BUILD_PYTORCH_USING_LIBTORCH_WHL:
+    os.environ["BUILD_LIBTORCHLESS"] = "ON"
```
Can we use only 1 var BUILD_PYTORCH_USING_LIBTORCH_WHL everywhere instead of 
```
BUILD_PYTORCH_USING_LIBTORCH_WHL
BUILD_LIBTORCHLESS
```

PaliC replied:
It's two separate builds so not at the moment.

atalman replied:
It looks like these PR introduce following new flags:
```
BUILD_LIBTORCH_WHL
BUILD_PYTORCH_USING_LIBTORCH_WHL
BUILD_LIBTORCHLESS
BUILD_PYTHON_ONLY
```

is it possible to define what each flag is used for? and maybe use fewer of the flags ?
I see following:
```
BUILD_PYTORCH_USING_LIBTORCH_WHL = os.getenv("BUILD_PYTHON_ONLY", "0") == "1"
...
 if BUILD_PYTORCH_USING_LIBTORCH_WHL:
        os.environ["BUILD_LIBTORCHLESS"] = "ON"
```
Does it mean we can use 
```
BUILD_PYTHON_ONLY = os.getenv("BUILD_PYTHON_ONLY", "0") == "1"
```
?

PaliC committed:
Update

[ghstack-poisoned]

PaliC commented:
> Please have CMakeLists.txt formatting changes in a separate PR.

Pretty sure lint would break if that was done as the only formatting here is done by `lintrunner -a -m "origin/main"`

PaliC committed:
Update

[ghstack-poisoned]

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>)
+  set(C10_LIB c10)
```
From what I understand:
```
if(NOT BUILD_LIBTORCHLESS)
```
is default path, however `` set(C10_LIB c10)`` is new line here.
Is this correct?

PaliC replied:
Yes. It helps reduce the if else statements in the CMake files. If I  could use C10_LIB instead of c10, then I did.

atalman replied:
Do you have an example of this use case ?

kit1980 commented:
> > Please have CMakeLists.txt formatting changes in a separate PR.
> 
> Pretty sure lint would break if that was done as the only formatting here is done by `lintrunner -a -m "origin/main"`

Then maybe submit the formatting first in a separate PR?
Right now it's very hard to see the functional changes vs formatting changes.

atalman commented on c10/cuda/CMakeLists.txt:
```diff
@@ -47,36 +51,42 @@ set(C10_CUDA_HEADERS
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../..>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>)
+  set(C10_CUDA_LIB c10_cuda)
```
Why are we not doing this at the beginning of the file ? So that everywhere else its 
```
${C10_CUDA_LIB}
```

PaliC replied:
https://github.com/pytorch/pytorch/pull/125249#pullrequestreview-2043971589 describes the purpose of these aliases as well.

atalman commented on c10/cuda/test/CMakeLists.txt:
```diff
@@ -15,7 +15,7 @@ if(BUILD_TEST)
     set(test_name "c10_cuda_${test_file_name}")
     add_executable(${test_name} "${test_src}")
-    target_link_libraries(${test_name} c10_cuda gtest_main)
+    target_link_libraries(${test_name} ${C10_CUDA_LIB} ${C10_LIB} gtest_main)
```
This change will affect both new wheels and old ? before we used to link only with
``
c10_cuda
`` 
now with both 
``
c10_cuda c10
``

PaliC replied:
c10_cuda contains c10, so this is effectively a no-op.

The error that led to this is that we were missing symbols in c10, so we explicitly link it here. My guess is that when we link from the external wheel, the associative libraries are not included (in this case you can't get to c10 from c10_cuda).

atalman replied:
Here can we add an if statement like this to avoid changing the default current build ?
```
if(BUILD_LIBTORCHLESS)
target_link_libraries(${test_name} ${C10_CUDA_LIB} ${C10_LIB} gtest_main)
else()
target_link_libraries(${test_name} ${C10_CUDA_LIB} gtest_main)
endif()

atalman reviewed (approved):
tentatively approving, please fix the lint

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-  target_compile_options(c10 PRIVATE "-fvisibility=hidden")
+if(BUILD_LIBTORCHLESS)
+  find_library(C10_LIB c10 PATHS $ENV{LIBTORCH_LIB_PATH} NO_DEFAULT_PATH)
 endif()
```
```else()
  set(C10_LIB c10)
endif()
```

This will be cleaner if we use C10_LIB everywhere in this case, not mix of c10 and C10_LIB

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+          util/*.h
+        )
+if(NOT BUILD_LIBTORCHLESS)
+  add_library(c10 ${C10_SRCS} ${C10_HEADERS})
```
```suggestion
  add_library(${C10_LIB}  ${C10_SRCS} ${C10_HEADERS})
```

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  add_library(c10 ${C10_SRCS} ${C10_HEADERS})
+  target_compile_options_if_supported(c10 "-Wdeprecated")
+  if(HAVE_SOVERSION)
+    set_target_properties(c10 PROPERTIES
```
```suggestion
    set_target_properties(${C10_LIB} PROPERTIES
```

and the other places where we pass c10 lib

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/../>
+      $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
+      $<INSTALL_INTERFACE:include>)
+  set(C10_LIB c10)
```
```suggestion
```

atalman commented on c10/CMakeLists.txt:
```diff
@@ -12,120 +12,127 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+  set(C10_LIB c10)
+endif()
 
 add_subdirectory(test)
```
Can we move lines 137...152 before ``if(NOT BUILD_LIBTORCHLESS)`` case. Then we can unify both ``if(NOT BUILD_LIBTORCHLESS)`` statements

atalman commented on c10/xpu/CMakeLists.txt:
```diff
@@ -4,6 +4,10 @@
 
 include(../../cmake/public/xpu.cmake)
```
We would need to add ``ciflow/xpu`` in order to test this build

atalman commented on caffe2/CMakeLists.txt:
```diff
@@ -789,6 +789,26 @@ set(Caffe2_CPU_SRCS ${Caffe2_CPU_SRCS_NON_AVX} ${Caffe2_CPU_SRCS_AVX2} ${Caffe2_
+  endif()
+  add_subdirectory(../torch torch)
+
+endif()
```
Here we probably would need to have:
```
else()
 set(TORCH_LIB, torch)
endif()
```

atalman commented on functorch/CMakeLists.txt:
```diff
@@ -16,7 +16,11 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE TORCH_API_INCLUDE_EXTENSION_H
+if(BUILD_LIBTORCHLESS)
+  target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIB} torch_python)
+else()
+  target_link_libraries(${PROJECT_NAME} PRIVATE torch torch_python)
```
We can probably replace these if/else by setting TORCH_LIB in advance see comment https://github.com/pytorch/pytorch/pull/126328/files#r1611944303

atalman commented on torch/CMakeLists.txt:
```diff
@@ -359,7 +365,9 @@ endif()
   target_compile_definitions(torch_python PRIVATE "-DBUILD_ONEDNN_GRAPH")
-  target_compile_definitions(torch_cpu PRIVATE "-DBUILD_ONEDNN_GRAPH")
+  if(NOT BUILD_LIBTORCHLESS)
+    target_compile_definitions(torch_cpu PRIVATE "-DBUILD_ONEDNN_GRAPH")
```
Is this failing with ``BUILD_LIBTORCHLESS`` on ? What is the failure we might want to followup on this

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update on "[Split Build] Add option to create libtorch wheel and use it to build pytorch as a separate wheel"


Creates an option to just build the libtorch portion of pytorch such that we have the necessary .so files.  Then it builds a torch package using the libtorch wheel. These options are enabled using ` BUILD_LIBTORCH_WHL` and `BUILD_PYTHON_ONLY`.

We run 

```
 BUILD_LIBTORCH_WHL=1 python setup.py install
python setup.py clean
BUILD_PYTHON_ONLY=1 python setup.py install
```

to produce

```
sahanpdevgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ls /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/torch/lib/                                                                                                                (pytorch-3.10) 
libshm.so*  libtorch_global_deps.so*  libtorch_python.so*
sahanpdevgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ldd build/lib/libtorch_python.so                                                                                                                                                                (pytorch-3.10) 
        linux-vdso.so.1 (0x00007ffdc2d37000)
        libtorch.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch.so (0x00007f539fe99000)
        libshm.so => /home/sahanp/pytorch/build/lib/libshm.so (0x00007f539fe90000)
        libcudnn.so.8 => /usr/local/cuda-12.1/targets/x86_64-linux/lib/libcudnn.so.8 (0x00007f539e800000)
        libnvToolsExt.so.1 => /usr/local/cuda/lib64/libnvToolsExt.so.1 (0x00007f539e400000)
        libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f539e000000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f539fda5000)
        libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f539ebe5000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f539dc00000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f539fea0000)
        libtorch_cpu.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch_cpu.so (0x00007f5392400000)
        libtorch_cuda.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libtorch_cuda.so (0x00007f5380000000)
        librt.so.1 => /lib64/librt.so.1 (0x00007f539fd9e000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f539fd99000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007f539fd94000)
        libc10.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libc10.so (0x00007f539eb07000)
        libmkl_intel_lp64.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_intel_lp64.so.2 (0x00007f537ec00000)
        libmkl_gnu_thread.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_gnu_thread.so.2 (0x00007f537ce00000)
        libmkl_core.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_core.so.2 (0x00007f5378800000)
        libomp.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/libomp.so (0x00007f539e707000)
        libcupti.so.12 => /usr/local/cuda/lib64/libcupti.so.12 (0x00007f5377e00000)
        libcudart.so.12 => /usr/local/cuda/lib64/libcudart.so.12 (0x00007f5377a00000)
        libc10_cuda.so => /home/sahanp/.conda/envs/pytorch-3.10/lib/python3.10/site-packages/libtorch/lib/libc10_cuda.so (0x00007f539ea6a000)
        libcusparse.so.12 => /usr/local/cuda/lib64/libcusparse.so.12 (0x00007f5368400000)
        libcufft.so.11 => /usr/local/cuda/lib64/libcufft.so.11 (0x00007f535ee00000)
        libcusolver.so.11 => /usr/local/cuda/lib64/libcusolver.so.11 (0x00007f534c800000)
        libcurand.so.10 => /usr/local/cuda/lib64/libcurand.so.10 (0x00007f5346200000)
        libcublas.so.12 => /usr/local/cuda/lib64/libcublas.so.12 (0x00007f533f800000)
        libcublasLt.so.12 => /usr/local/cuda/lib64/libcublasLt.so.12 (0x00007f531e800000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007f539ea63000)
        libnvJitLink.so.12 => /usr/local/cuda/lib64/libnvJitLink.so.12 (0x00007f531b800000)
sahanpdevgpu086 ~/pytorch (detached HEAD|REBASE-i 3/5)> ldd build/lib/libtorch_global_deps.so                                                                                                                                                           (pytorch-3.10) 
        linux-vdso.so.1 (0x00007ffc265df000)
        libmkl_intel_lp64.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_intel_lp64.so.2 (0x00007fa93fc00000)
        libmkl_gnu_thread.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_gnu_thread.so.2 (0x00007fa93de00000)
        libmkl_core.so.2 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libmkl_core.so.2 (0x00007fa939800000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fa940f05000)
        libcudart.so.12 => /usr/local/cuda/lib64/libcudart.so.12 (0x00007fa939400000)
        libnvToolsExt.so.1 => /usr/local/cuda/lib64/libnvToolsExt.so.1 (0x00007fa939000000)
        libgomp.so.1 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libgomp.so.1 (0x00007fa93fb07000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fa938c00000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fa940efe000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fa940ef9000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fa940ff5000)
        librt.so.1 => /lib64/librt.so.1 (0x00007fa940ef2000)
        libstdc++.so.6 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libstdc++.so.6 (0x00007fa93921d000)
        libgcc_s.so.1 => /home/sahanp/.conda/envs/pytorch-3.10/lib/libgcc_s.so.1 (0x00007fa93faec000)
        ```

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC committed:
Update

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge

PaliC committed:
Update

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge
